.github/workflows/unittest.yml

```
@@ -29,7 +29,7 @@ jobs:
       - name: run unit tests
         run: |
           echo "running unit tests"
-          python -m pytest --cov=src --cov-report=xml --cov-report=term-missing tests/
+          python -m pytest --cov=src --cov-report=xml --cov-report=term-missing -m "not cuda_required" tests/
```

since the pytest.ini information is in the `pyproject.toml`, you should be able to drop the directory
```suggestion
          python -m pytest --cov=src --cov-report=xml --cov-report=term-missing -m "not cuda_required"
```